api-server/api-server-common/src/storage/impls/in_memory/mod.rs

```diff
@@ -0,0 +1,174 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+pub mod transactional;
+
+pub const CURRENT_STORAGE_VERSION: u32 = 1;
+
+use std::collections::BTreeMap;
+
+use common::{
+    chain::{Block, ChainConfig, GenBlock, SignedTransaction, Transaction},
+    primitives::{BlockHeight, Id},
+};
+
+use crate::storage::storage_api::{block_aux_data::BlockAuxData, ApiStorageError};
+
+pub struct ApiInMemoryStorage {
```

Comment:
I find the `Api` prefix in the name a bit weird, since it actually implements the database. Something with `Api` in name I'd expect to represent merely an interface.

Reply:
I understand... though this is at worst temporary because honestly I don't know what to call this and I need to progress... ApiServer is better?

Reply:
>  ApiServer is better?

Yes

Reply:
Done